### PR TITLE
RenderTexture now checks whether the display object its rendering has children.

### PR DESCRIPTION
**Pull request by adelciotto**
RenderTexture now checks whether the display object its rendering has children or not.

Issue: https://github.com/GoodBoyDigital/pixi.js/issues/1537

### CODE REVIEW

**Comment by englercj** [comment]
@GoodBoyDigital Looks good to me, thoughts?

File: src/core/textures/RenderTexture.js [hunk 1]
```diff
@@ -270,11 +270,18 @@ RenderTexture.prototype.renderWebGL = function (displayObject, matrix, clear, up
 
         // Time to update all the children of the displayObject with the new matrix..
         var children = displayObject.children;
-        var i, j;
+        if (children)
```
Comment (drkibitz) [inline src/core/textures/RenderTexture.js]: Do you mean for this condition to allow for `children.length == 0` ?

File: src/core/textures/RenderTexture.js [hunk 2]
```diff
@@ -322,11 +329,18 @@ RenderTexture.prototype.renderCanvas = function (displayObject, matrix, clear, u
 
     // Time to update all the children of the displayObject with the new matrix..
     var children = displayObject.children;
-    var i, j;
+    if (children)
```
Comment (drkibitz) [inline src/core/textures/RenderTexture.js]: Do you mean for this condition to allow for `children.length == 0` ?

**Comment by englercj** [comment]
Good catch @drkibitz I missed that.

**Comment by adelciotto** [comment]
yeah cheers @drkibitz, silly mistake by me.

**Comment by GoodBoyDigital** [comment]
hey there! thanks for this PR @adelciotto . Im not 100% sure what problem this is solving mind? the displayObject being rendered should not have it own transform called as it would override the matrix passed to it IYSWIM?

**Comment by adelciotto** [comment]
Hey @GoodBoyDigital, I do see what you mean, I was attempting to solve this issue: https://github.com/GoodBoyDigital/pixi.js/issues/1537 . Although now I think what I have done here may be pointless, as DisplayObject is an abstract base class, all the renderable objects in the framework like Text, Sprite, Graphics etc all inherit from container, and therefore will always have a defined children array. I believe if something simply extends DisplayObject and and size of it is changed and is rendered to a RenderTexture, the boundaries won't update. Please correct me though if i'm wrong though, I'm still very much learning my way around the framework.

**Comment by englercj** [comment]
@GoodBoyDigital Looks like you assigned the linked bug to yourself, so assigning this PR to you as well to do with as you will :)

**Comment by GoodBoyDigital** [comment]
I think v3 should have all this stuff licked so closing for now. Thanks @adelciotto

**Pull request closed without merging**